IS0402Test.py

```diff
@@ -2073,14 +2056,11 @@ def post_subscription(self, test, sub_json, query_url=None):
         except json.decoder.JSONDecodeError:
             raise NMOSTestException(test.FAIL("Non-JSON response returned for Query API subscription request"))
 
-    def post_resource(self, test, type, data=None, reg_url=None, codes=None):
+    def post_resource(self, test, type, data=None, reg_url=None, codes=None, fail=Test.FAIL):
```

Comment:
Generally looks good, I'm just a bit uncertain how the default 'fail' argument works for this method and the next?

Reply:
I missed the override of the `self` argument in `raise NMOSTestException(fail(test, ` which threw me as I thought the method on the class was being called without initialisation.